plugins/input/postgis/postgis_datasource.cpp

```diff
@@ -482,21 +507,18 @@ std::string postgis_datasource::sql_bbox(box2d<double> const& env) const
 {
     std::ostringstream b;
 
-    if (srid_ > 0)
-    {
-        b << "ST_SetSRID(";
-    }
-
-    b << "'BOX3D(";
+    b << "ST_MakeEnvelope(";
     b << std::setprecision(16);
-    b << env.minx() << " " << env.miny() << ",";
-    b << env.maxx() << " " << env.maxy() << ")'::box3d";
```

Comment:
No, going through box3d is actually a little more involved (build box3d type, cast to geometry, then use) than building a geometry in one go with `ST_MakeEnvelope()`